dask/bag/core.py

```diff
@@ -7,19 +7,25 @@
 import os
 import uuid
 from warnings import warn
+from distutils.version import LooseVersion
 
 from ..utils import ignoring
 
 from toolz import (merge, take, reduce, valmap, map, partition_all, filter,
-        remove, compose, curry)
+                   remove, compose, curry, first, second, accumulate)
 from toolz.compatibility import iteritems, zip
 import toolz
+_implement_accumulate = LooseVersion(toolz.__version__) > '0.7.4'
```

Comment:
I get that there is complexity here, but it seems better to keep the
complexity close to the source of the reason for the complexity.  There is
some small cost to the redirection.  There is also some desire to isolate
the top level from special case stuff.

On Thu, Jun 2, 2016 at 10:30 AM, Jim Crist <notifications@github.com> wrote:

> In dask/bag/core.py
> <https://github.com/dask/dask/pull/1201#discussion_r65583173>:
>
> >  from toolz.compatibility import iteritems, zip
> >  import toolz
> > +_implement_accumulate = LooseVersion(toolz.__version__) > '0.7.4'
>
> The trick is we need different versions for either cytoolz or pytoolz, and
> accumulate may come from either (or none) of them. It's a lot of work to
> get around the versioning shenanigans. Can nix all of this once
> dependencies are updated.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/dask/dask/pull/1201/files/83968ce8c0b81b8f01ccf0fc0b75524ed91f695d#r65583173>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AASszO2wFLIMGJ3OfCKTCCqtHfcAXH1Lks5qHxMmgaJpZM4IryYK>
> .
>